app/shared/ast/src/constants.ts

```diff
@@ -4,27 +4,194 @@ export const ECMA_VERSION = 11;
+  FunctionDeclaration = 'FunctionDeclaration',
+  ExportDefaultDeclaration = 'ExportDefaultDeclaration',
+  VariableDeclarator = 'VariableDeclarator',
   // Expression - https://github.com/estree/estree/blob/master/es5.md#expressions
-  MemberExpression = "MemberExpression",
-  FunctionExpression = "FunctionExpression",
-  ArrowFunctionExpression = "ArrowFunctionExpression",
-  ObjectExpression = "ObjectExpression",
-  ArrayExpression = "ArrayExpression",
-  ThisExpression = "ThisExpression",
+  MemberExpression = 'MemberExpression',
+  FunctionExpression = 'FunctionExpression',
+  ArrowFunctionExpression = 'ArrowFunctionExpression',
+  ObjectExpression = 'ObjectExpression',
+  ArrayExpression = 'ArrayExpression',
+  ThisExpression = 'ThisExpression',
 }
+
+export const JAVASCRIPT_KEYWORDS = {
+  Array: 'Array',
+  await: 'await',
+  Boolean: 'Boolean',
+  break: 'break',
+  case: 'case',
+  catch: 'catch',
+  class: 'class',
+  const: 'const',
+  continue: 'continue',
+  Date: 'Date',
+  debugger: 'debugger',
+  default: 'default',
+  delete: 'delete',
+  do: 'do',
+  else: 'else',
+  enum: 'enum',
+  eval: 'eval',
+  export: 'export',
+  extends: 'extends',
+  false: 'false',
+  finally: 'finally',
+  for: 'for',
+  function: 'function',
+  Function: 'Function',
+  hasOwnProperty: 'hasOwnProperty',
+  if: 'if',
+  implements: 'implements',
+  import: 'import',
+  in: 'in',
+  Infinity: 'Infinity',
+  instanceof: 'instanceof',
+  interface: 'interface',
+  isFinite: 'isFinite',
+  isNaN: 'isNaN',
+  isPrototypeOf: 'isPrototypeOf',
+  JSON: 'JSON',
+  length: 'length',
+  let: 'let',
+  Math: 'Math',
+  name: 'name',
+  NaN: 'NaN',
+  new: 'new',
+  null: 'null',
+  Number: 'Number',
+  Object: 'Object',
+  package: 'package',
+  private: 'private',
+  protected: 'protected',
+  public: 'public',
+  return: 'return',
+  static: 'static',
+  String: 'String',
+  super: 'super',
+  switch: 'switch',
+  this: 'this',
+  throw: 'throw',
+  toString: 'toString',
+  true: 'true',
+  try: 'try',
+  typeof: 'typeof',
+  undefined: 'undefined',
+  valueOf: 'valueOf',
+  var: 'var',
+  void: 'void',
+  while: 'while',
+  with: 'with',
+  yield: 'yield',
+};
+
+export const WINDOW_OBJECT_PROPERTIES = {
+  closed: 'closed',
+  console: 'console',
+  defaultStatus: 'defaultStatus',
+  document: 'document',
+  frameElement: 'frameElement',
+  frames: 'frames',
+  history: 'history',
+  innerHeight: 'innerHeight',
+  innerWidth: 'innerWidth',
+  length: 'length',
+  localStorage: 'localStorage',
+  location: 'location',
+  name: 'name',
+  navigator: 'navigator',
+  opener: 'opener',
+  outerHeight: 'outerHeight',
+  outerWidth: 'outerWidth',
+  pageXOffset: 'pageXOffset',
+  pageYOffset: 'pageYOffset',
+  parent: 'parent',
+  screen: 'screen',
+  screenLeft: 'screenLeft',
+  screenTop: 'screenTop',
+  screenY: 'screenY',
+  scrollX: 'scrollX',
+  scrollY: 'scrollY',
+  self: 'self',
+  status: 'status',
+  top: 'top',
+};
```

Comment:
```suggestion
export const WINDOW_OBJECT_PROPERTIES = {
  console: 'console',
  localStorage: 'localStorage',
  name: 'name',
};
```
As the evaluations in happening in worker, we can avoid DOM APIs